utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploymentScripts/Copy-VhdToStorageAccount.ps1

```diff
@@ -0,0 +1,117 @@
+
+    Copy a VHD baked by image template 'vhd-img-template-001-2022-07-29-15-54-01' in resource group 'validation-rg' to destination storage account 'vhdstorage001' in a blob container named 'vhds' and wait for the completion of the copy. Save the VHD file as 'vhd-img-template-001.vhd'.
+#>
+
+[CmdletBinding(SupportsShouldProcess)]
+param (
+    [Parameter(Mandatory = $true)]
+    [string] $ImageTemplateName,
+
+    [Parameter(Mandatory = $true)]
+    [string] $ImageTemplateResourceGroup,
+
+    [Parameter(Mandatory = $true)]
+    [string] $DestinationStorageAccountName,
+
+    [Parameter(Mandatory = $false)]
+    [string] $DestinationContainerName = 'vhds',
+
+    [Parameter(Mandatory = $false)]
+    [string] $VhdName = $ImageTemplateName,
+
+    [Parameter(Mandatory = $false)]
+    [switch] $WaitForComplete
+)
+
+begin {
+    Write-Debug ('{0} entered' -f $MyInvocation.MyCommand)
+
+    # Install required modules
+    $currentVerbosePreference = $VerbosePreference
+    $VerbosePreference = 'SilentlyContinue'
+    $requiredModules = @(
+        'Az.ImageBuilder',
+        'Az.Storage'
+    )
+    foreach ($moduleName in $requiredModules) {
+        if (-not ($installedModule = Get-Module $moduleName -ListAvailable)) {
+            Install-Module $moduleName -Repository 'PSGallery' -Force -Scope 'CurrentUser'
+            if ($installed = Get-Module -Name $moduleName -ListAvailable) {
+                Write-Verbose ('Installed module [{0}] with version [{1}]' -f $installed.Name, $installed.Version) -Verbose
+            }
+        } else {
+            Write-Verbose ('Module [{0}] already installed in version [{1}]' -f $installedModule[0].Name, $installedModule[0].Version) -Verbose
+        }
+    }
+    $VerbosePreference = $currentVerbosePreference
+}
+
+process {
+    # Retrieving and initializing parameters before the blob copy
+    Write-Verbose 'Initializing source storage account parameters before the blob copy' -Verbose
+    $imgtRunOutput = Get-AzImageBuilderRunOutput -ImageTemplateName $imageTemplateName -ResourceGroupName $imageTemplateResourceGroup | Where-Object ArtifactUri -NE $null
+    $sourceUri = $imgtRunOutput.ArtifactUri
+    $sourceStorageAccountName = $sourceUri.Split('//')[1].Split('.')[0]
+    $sourceStorageAccount = Get-AzStorageAccount | Where-Object StorageAccountName -EQ $sourceStorageAccountName
+    $sourceStorageAccountContext = $sourceStorageAccount.Context
+    $sourceStorageAccountRGName = $sourceStorageAccount.ResourceGroupName
+    Write-Verbose ('Retrieving artifact uri [{0}] stored in resource group [{1}]' -f $sourceUri, $sourceStorageAccountRGName) -Verbose
+
+    Write-Verbose 'Initializing destination storage account parameters before the blob copy' -Verbose
+    $destinationStorageAccount = Get-AzStorageAccount | Where-Object StorageAccountName -EQ $destinationStorageAccountName
```

Comment:
Minor thing, but I'd suggest to fetch the storage accounts only once in the script and then filter by name where you need it

Reply:
Updated